### ISSUE TITLE
Initial examples lost

### ISSUE THREAD

**Issue by JohnLangford** [Microsoft/mwt-ds#27]
When I start entering information immediately after successful deployment and before the trainer is started the examples are apparently lost.  For foo-7 and foo-9 in the current deployment, I put 20 or 30 examples into things initially.  After the trainer started several minutes later it said "Trainer OK. Total learned examples: 2" and "Trainer OK. Total learned examples: 3".   

Why is information being lost?  And can we stop the information loss?

**Comment by JohnLangford** [comment]
It's probably worthwhile to also have a line that says "Trainer not started" until we know otherwise.

**Comment by JohnLangford** [comment]
(This was 2/3 deployments.  5/8 failed template expansion this morning.)

**Comment by eisber** [comment]
TL;DR Trainer needs to start reading events from the beginning of the event hub stream on first start.

Events are present in the pipeline (pre- and post join query), but since the trainer started after the ASA join query produced events and the trainer starts reading from DateTime.UtcNow on a fresh start (= no checkpoint available) the trainer did not consider the events.
Online Trainer Version 8.2.0.5 will contain a fix.
